Using NPM-prefixed Import Syntax Breaks Lock File

## Deno Info
Version: Deno 2.0.3

## Steps to reproduce

File Tree:
```tree
- .vscode
  - settings.json
- app.ts
- deno.json
```

Deno.json:
```json
{
    "imports": {
        "@aws-sdk/client-s3": "npm:@aws-sdk/client-s3@^3.679.0"
    }
}
```

1. Add the following to `app.ts`:
   ```ts
   import { S3Client } from '@aws-sdk/client-s3';
   console.log(S3Client);
   ```
2. Run `deno run -A app.ts`:
   Console Output: `[class S3Client extends Client]`
3. Update `app.ts`:
   ```ts
   import { S3Client } from 'npm:@aws-sdk/client-s3';
   console.log(S3Client);
   ```
4. Run `deno run -A app.ts`:
   Console Output: `error: Could not resolve 'npm:@aws-sdk/client-s3@3.679.0'.`
5. Update `app.ts` to original:
   ```ts
   import { S3Client } from '@aws-sdk/client-s3';
   console.log(S3Client);
   ```
6. Run `deno run -A app.ts`:
   Console Output: `error: Could not resolve 'npm:@aws-sdk/client-s3@3.679.0'.`

## Expected Behavior

Both import syntaxes should work.
```
import { S3Client } from '@aws-sdk/client-s3';
import { S3Client } from 'npm:@aws-sdk/client-s3';
```

## Notes

The lock file seems to become “corrupted” once the command in step 4 is run.

I was able to reproduce with `2.0.2` and `2.0.3`.

`main.ts` file

```typescript
import { S3Client } from "npm:@aws-sdk/client-s3@^3.679.0";
console.log(S3Client);
```

No `deno.json` yet. No `deno.lock` yet.

Then run `deno run -A -r main.ts`.

Outputs `[class S3Client extends Client]`. 

Now let's create an import map. Create file with `touch deno.json`. Then run `deno add 'npm:@aws-sdk/client-s3@^3.679.0'`. This generates `deno.lock` and `deno.json`. 

`deno.json` will now have an import map.
```json
{
  "imports": {
    "@aws-sdk/client-s3": "npm:@aws-sdk/client-s3@^3.682.0"
  }
}
```
Run `deno run -A main.ts`. Gets error.

```
error: Could not resolve 'npm:@aws-sdk/client-s3@3.682.0'.

Caused by:
    [ERR_MODULE_NOT_FOUND] Cannot find module 'file:///home/uncomfy/development/projects/.cache/deno/npm/registry.npmjs.org/@aws-sdk/client-s3/3.682.0_1/index.js' imported from 'file:///tmp/s3/main.ts'
```

Then let's try removing the lockfile with `rm deno.lock`

Run `deno run -A main.ts` or `deno run -A -r main.ts`.

Outputs `[class S3Client extends Client]`.  This also regenerates the lockfile... of which I think points to the import from `main.ts` and not from `deno.json`. I wanted to try to test this theory so I ran `deno install`. The lockfile was regenerated. 

Now it points to the import map in `deno.json`. The error came back.

> [!NOTE]
> Notice the `3.682.0_1`. By copying the `3.682.0` directory to `3.682.0_1`, it removes the error.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -60,7 +60,7 @@ deno_graph = "=0.89.2"
 deno_lint = "=0.74.0"
 deno_lockfile = "=0.25.0"
 deno_media_type = { version = "=0.2.8", features = ["module_specifier"] }
-deno_npm = "=0.28.0"
+deno_npm = "=0.29.0"
 deno_path_util = "=0.3.2"
 deno_permissions = { version = "0.55.0", path = "./runtime/permissions" }
 deno_runtime = { version = "0.204.0", path = "./runtime" }

